prediction_market_agent/agents/safe_guard_agent/app.py

```diff
@@ -22,15 +23,24 @@
 - Test out validation in this demo, without adding agent as a signer or doing any real transactions
 - Call the validation via endpoint and see the documentation for it
 - See the list of deployed agents that can be added as signers to your Safe
-
-*Currently works only on Gnosis Chain.*
 """
 )
 
+chain_name, chain_id = st.selectbox(
+    "Select chain",
+    [("Ethereum", ETHEREUM_ID), ("Gnosis", GNOSIS_CHAIN_ID)],
+    index=1,
+    format_func=lambda x: x[0],
+)
+
 pages = [
-    st.Page(demo_page.demo_page, title="Demo", url_path="demo"),
-    st.Page(api_page.api_page, title="API", url_path="api"),
-    st.Page(agents_page.agents_page, title="Agents", url_path="agents"),
+    st.Page(demo_page.get_demo_page(chain_id=chain_id), title="Demo", url_path="demo"),
+    st.Page(api_page.get_api_page(chain_id=chain_id), title="API", url_path="api"),
+    st.Page(
+        agents_page.get_agents_page(chain_id=chain_id),
+        title="Agents",
+        url_path="agents",
+    ),
```

Comment:
I don't like this hard separation too much - in theory, we could easily support all 19 networks where the Transaction Service operates (https://docs.safe.global/advanced/smart-account-supported-networks?service=Transaction+Service&version=v1.4.1).
In that case, we would have 19 pages in the app, which wouldn't be very user-friendly.
I would expect instead to have 1 page (with the chain dropdown) that simply is passed as a parameter to the underlying function.

Reply:
> I would expect instead to have 1 page (with the chain dropdown) that simply is passed as a parameter to the underlying function.

But that's exactly how it is right now..

Reply:
```
pages = [
    st.Page(demo_page.get_demo_page(chain_id=chain_id), title="Demo", url_path="demo"),
    st.Page(api_page.get_api_page(chain_id=chain_id), title="API", url_path="api"),
    st.Page(
```
Ow I thought this meant having 1 page per chain. My bad for not following everything Streamlit offers. Closing this.